src/components/EmojiReactions/EmojiReactions.tsx

```diff
@@ -0,0 +1,286 @@
+        if (userReactions && userReactions.usedBy.includes(clientId)) {
+          userReactions.usedBy.splice(userReactions.usedBy.indexOf(clientId), 1)
+          usedEmojiCollection[usedEmojiCollection.indexOf(userReactions)] =
+            userReactions
+        }
+        break
+    }
+    return usedEmojiCollection
+  }
+
+  const updateMessageFromHistory = (
+    messageIndex: number,
+    history: Types.PaginatedResult<Types.Message>
+  ) => {
+    const lastPublishedMessage = history?.items[messageIndex]
+    // Get reactions of the published message
+    if (messageIndex > 0) {
+      for (let i = messageIndex - 1; i >= 0; i--) {
+        const emoji = history?.items[i].data.body
+        const client = history?.items[i].clientId
+        const event = history?.items[i].name
+
+        if (usedEmojiCollection.length > 0) {
+          for (const usage of usedEmojiCollection) {
+            updateEmojiCollection(emoji, client, event)
+          }
+        } else {
+          const emojiUse: EmojiUsage = { usedBy: [client], emoji: emoji }
+          usedEmojiCollection.push(emojiUse)
+        }
+      }
+    }
+    // Update chat message
+    setChatMessage({
+      author: lastPublishedMessage?.data.author,
+      content: lastPublishedMessage?.data.content,
+      timeserial: lastPublishedMessage?.id,
+      reactions: usedEmojiCollection,
+    })
+  }
+
+  useEffect(() => {
+    setProjectInfo({
+      name: 'Emoji Reactions',
+      repoNameAndPath:
+        'atomic-examples/tree/main/src/components/MessageReactions',
+      topic: 'emoji-reactions',
+    })
+  }, [])
+
+  useEffect(() => {
+    // Subscribe to message reactions
+    getMessageReactions()
+    // Keep last published message and reactions
+    channel.history((err, result) => {
+      // Get index of last sent message from history
+      const lastPublishedMessageIndex: any = result?.items.findIndex(
+        (message) => message.name == SEND_EVENT
+      )
+
+      if (lastPublishedMessageIndex >= 0) {
+        updateMessageFromHistory(lastPublishedMessageIndex, result!)
+      } else {
+        // Load random message when no sent message history
+        sendMessage()
+      }
+    })
+  }, [])
+
+  return (
+    <div className="p-6">
+      <h1>Emoji Reactions</h1>
+      <p>
+        Hello to you and welcome to Ably. You can react to any of these
+        messages.
+      </p>
+      <p>Go ahead, give it a try! You can always checkout the source code </p>
+      {chatMessage.author ? (
+        <div className="p-6 max-w-fit mx-auto bg-slate-50 rounded-xl shadow-lg flex items-center space-x-4 m-6">
+          <img className="bg-gradient-to-r from-cyan-500 to-blue-500 h-12 w-12 rounded-full mb-2 shrink-0"></img>
+          <div>
+            <p className="text-sm font-small text-black">
+              {chatMessage.author}
+            </p>
+            <p className="text-slate-500"> {chatMessage.content} </p>
+            <div className="flex flex-row flex-wrap">
```

Comment:
I think we need to put a few comments into the HTML since we're expecting people to read it.

If I understand this block displays and lets you interact with reactions already sent by you and others.